circuits/circom/core/qv/tallyVotes.circom

```diff
@@ -0,0 +1,310 @@
+    var numVoteOptions = TREE_ARITY ** voteOptionTreeDepth;
+
+    // Number of elements in each ballot.
+    var BALLOT_LENGTH = 2;
+    // Index for the nonce in the ballot array.
+    var BALLOT_NONCE_IDX = 0;
+    // Index for the voting option root in the ballot array.
+    var BALLOT_VO_ROOT_IDX = 1;
+    // Difference in tree depths, used in path calculations.
+    var k = stateTreeDepth - intStateTreeDepth;
+
+    // Root of the state Merkle tree, representing the overall state before voting.
+    signal input stateRoot;
+    // Root of the ballot Merkle tree, representing the submitted ballots.
+    signal input ballotRoot;
+    // Salt used in commitment to secure the ballot data.
+    signal input sbSalt;
+
+    // Inputs combined into a hash to verify the integrity and authenticity of the data.
+    signal input packedVals;
+    // Commitment to the state and ballots.
+    signal input sbCommitment;
+    // Commitment to the current tally before this batch.
+    signal input currentTallyCommitment;
+    // Commitment to the new tally after processing this batch.
+    signal input newTallyCommitment;
+    
+    // A tally commitment is the hash of the following salted values:
+    //   - the vote results,
+    //   - the number of voice credits spent per vote option,
+    //   - the total number of spent voice credits.
+
+    // Hash of all inputs to ensure they are unchanged and authentic.
+    signal input inputHash;
+    
+    // Ballots and their corresponding path elements for verification in the tree.
+    signal input ballots[batchSize][BALLOT_LENGTH];
+    signal input ballotPathElements[k][TREE_ARITY - 1];
+    signal input votes[batchSize][numVoteOptions];
+
+    // Current results for each vote option.
+    signal input currentResults[numVoteOptions];
+    // Salt for the root of the current results.
+    signal input currentResultsRootSalt;
+    
+    // Total voice credits spent so far.
+    signal input currentSpentVoiceCreditSubtotal;
+    // Salt for the total spent voice credits.
+    signal input currentSpentVoiceCreditSubtotalSalt;
+
+    // Spent voice credits per vote option.
+    signal input currentPerVOSpentVoiceCredits[numVoteOptions];
+    // Salt for the root of spent credits per option.
+    signal input currentPerVOSpentVoiceCreditsRootSalt;
+
+    // Salt for the root of the new results.
+    signal input newResultsRootSalt;
+    // Salt for the new spent credits per vote option root.
+    signal input newPerVOSpentVoiceCreditsRootSalt;
+    // Salt for the new total spent voice credits root. 
+    signal input newSpentVoiceCreditSubtotalSalt;
+    // The number of total registrations, used to validate the batch index.
+    signal numSignUps;
+    // Index of the first ballot in this batch.
+    signal batchStartIndex;
+
+    // Verify sbCommitment.
+    var sbCommitmentHash = PoseidonHasher(3)([stateRoot, ballotRoot, sbSalt]);
```

Comment:
Yes, it is a very good idea. I prefer `computed` instead of `calculated` since we are broadly speaking of more complex things instead of just arithmetics